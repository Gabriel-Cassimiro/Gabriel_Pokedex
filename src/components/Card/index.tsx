import React from "react"
import Image from "next/image"
import { PokeType } from "../../styles/pages/home.styles"
import { Poke } from "../../pages"

export interface CardProps {
	pokemon: Poke
}

const Card: React.FC<CardProps> = ({ pokemon }) => {
	return (
		<div className="PokemonSingle" key={pokemon.id}>
			<div className="image">
				<Image
					src={pokemon.img}
					alt={pokemon.name}
					height={150}
					width={150}
					layout={"fixed"}
				></Image>
			</div>

			<div className="pokeNum">#{pokemon.num}</div>
			<h2>{pokemon.name}</h2>
			<div className="types">
				{pokemon.type.map(types => {
					return (
						<PokeType key={types} color={types.toLowerCase()}>
							{types}
						</PokeType>
					)
				})}
			</div>
		</div>
	)
}

export default Card
