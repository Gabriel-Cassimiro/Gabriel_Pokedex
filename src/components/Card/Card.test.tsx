import { render, screen } from "@testing-library/react"

import Card, { CardProps } from "./index"
import { charmander } from "../../tests/mocks/pokemons.mock"
import { TestProvider } from "../../tests/utils"

describe("Test Card component", () => {
	function Element({ pokemon }: CardProps) {
		return (
			<TestProvider>
				<Card pokemon={pokemon} />
			</TestProvider>
		)
	}

	beforeEach(() => {
		jest.clearAllMocks()
	})

	it("should render correctly", () => {
		const { container } = render(<Element pokemon={charmander} />)

		expect(screen.getByText("Charmander")).toBeInTheDocument()

		expect(container).toMatchSnapshot()
	})
})
