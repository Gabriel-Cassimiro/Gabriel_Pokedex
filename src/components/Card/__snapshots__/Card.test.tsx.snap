// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Card component should render correctly 1`] = `
<div>
  <div
    class="PokemonSingle"
  >
    <div
      class="image"
    >
      <div
        style="overflow: hidden; box-sizing: border-box; display: inline-block; position: relative; width: 150px; height: 150px;"
      >
        <noscript />
        <img
          alt="Charmander"
          data-nimg="true"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </div>
    </div>
    <div
      class="pokeNum"
    >
      #
      004
    </div>
    <h2>
      Charmander
    </h2>
    <div
      class="types"
    >
      <div
        class="homestyles__PokeType-sc-1nzr84s-2 eYwIIJ"
        color="fire"
      >
        Fire
      </div>
    </div>
  </div>
</div>
`;
